src/OptionSettlement.sol

```diff
@@ -188,31 +188,52 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
 
     /// @inheritdoc IOptionSettlementEngine
     function write(uint256 optionId, uint112 amount) external returns (uint256 claimId) {
-        if (amount == 0) {
-            revert AmountWrittenCannotBeZero();
-        }
+        /// claimId == 0 signifies that a new claim NFT should be minted
+        return write(0, optionId, amount);
+    }
 
+    /// @inheritdoc IOptionSettlementEngine
+    function write(uint256 claimId, uint256 optionId, uint112 amount) public returns (uint256) {
         (uint160 _optionId, uint96 claimIndex) = getDecodedIdComponents(optionId);
+
+        // claim index must be zero in lower 96b for provided option Id
         if (claimIndex != 0) {
-            // claim index must be zero in lower 96b
             revert InvalidOption(optionId);
         }
 
-        Option storage optionRecord = _option[_optionId];
-
-        if (optionRecord.expiryTimestamp <= block.timestamp) {
-            revert ExpiredOption(optionId, optionRecord.expiryTimestamp);
+        // claim provided must match the option provided
+        if (claimId != 0 && (claimId >> 96) != (optionId >> 96)) {
+            revert EncodedOptionIdInClaimIdDoesNotMatchProvidedOptionId(claimId, optionId);
         }
 
-        uint256 rxAmount = amount * optionRecord.underlyingAmount;
-        uint256 fee = ((rxAmount / 10000) * feeBps);
-        address underlyingAsset = optionRecord.underlyingAsset;
+        Option storage optionRecord = _writeOptions(_optionId, amount);
+        uint256 mintClaimNft = 0;
+
+        if (claimId == 0) {
+            // create new claim
```

Comment:
In the case where a claim is provided, it must exist, elsewise the user must use the non overloaded version of this function.

Reply:
that's not the way it's written currently, so just want to describe what's there currently and see if you disagree with the design.

write(option, amount) writes `amount` of `option`, and puts this lot into a new `Claim`.

write(claimId, option, amount) writes `amount` of `option`, and if claimID is not zero, adds this amount to the existing claimID. If claimId is zero, then this is taken to mean "a claim does not exist, go create one".

write(option, amount) is just a call to [overloaded] write(0, option, amount)